rclcpp/src/rclcpp/duration.cpp

```diff
@@ -181,8 +182,10 @@ bounds_check_duration_scale(int64_t dns, double scale, uint64_t max)
 {
   auto abs_dns = static_cast<uint64_t>(std::abs(dns));
   auto abs_scale = std::abs(scale);
-
-  if (abs_scale > 1.0 && abs_dns > static_cast<uint64_t>(max / abs_scale)) {
+  long double max_ld = static_cast<long double>(max);
```

Comment:
Please avoid the one-time use variable and use the `static_cast` directly in the condition below.

Reply:
ok, changed